restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/util/RCUtils.java

```diff
@@ -230,37 +233,50 @@ public static boolean shouldRegisterForPush(HashMap<String, Object> parameters,
          throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_RESTCOMM_DOMAIN_MISSING);
       }
 
+      return false;
+
+   }
+
+   public static boolean shouldRegisterForPush(HashMap<String, Object> parameters, StorageManagerInterface storageManagerInterface) throws RCException{
+
+      validatePushSettings(parameters);
```

Comment:
Can we pull validatePushSettings() call out to RCDevice.registerForPushNotifications()? That way in shouldRegisterForPush() method we separate concerns and make sure that all it does is decide whether we should register for push, as it's name implies. Wdyt?